gpage.h

```diff
@@ -222,16 +228,17 @@ namespace gcpp {
 	//
 	inline
 	bool gpage::contains(gsl::not_null<const byte*> p) const noexcept {
-		return &storage[0] <= p && p < &storage[total_size - 1];
+		auto cmp = std::less<>{};
+		return !cmp(p, &storage[0]) && cmp(p, &storage[total_size - 1]);
```

Comment:
Looking at this now, it seems both the original and replacement code might miss the final byte (the original test probably should have been two `<=`). Also, in the other places you consistently replaced `&storage[0]` with `storage.get()`.

So perhaps `return !cmp(p, storage.get()) && cmp(p, storage.get() + total_size);` ?